docs/can-guides/commitment/recipes/canvas-clock/6-min-hour-hands.js

```diff
@@ -1,29 +1,37 @@
+import { Component } from "//unpkg.com/can@5/core.mjs";
+
 // 60 = 2π
 const base60ToRadians = (base60Number) =>
   2 * Math.PI * base60Number / 60;
 
-can.Component.extend({
+Component.extend({
   tag: "analog-clock",
-  view: '<canvas id="analog" width="255" height="255"></canvas>',
+  view: '<canvas this:to="canvasElement" id="analog" width="255" height="255"></canvas>',
   ViewModel: {
+    // the canvas element
+    canvasElement: HTMLCanvasElement,
+
+    // the canvas 2d context
+    get canvas() {
+      return this.canvasElement.getContext("2d");
+    },
+
+    drawNeedle(length, base60Distance, styles, center) {
+      Object.assign(this.canvas, styles);
+      const x = center + length * Math.sin(base60ToRadians(base60Distance));
+      const y = center + length * -1 * Math.cos(base60ToRadians(base60Distance));
+      this.canvas.beginPath();
+      this.canvas.moveTo(center, center);
+      this.canvas.lineTo(x, y);
+      this.canvas.closePath();
+      this.canvas.stroke();
+    },
+    
     connectedCallback(element) {
-      const canvas = element.firstChild.getContext("2d");
       const diameter = 255;
       const radius = diameter/2 - 5;
       const center = diameter/2;
 
-      const drawNeedle = (length, base60Distance, styles) => {
-        Object.assign(canvas, styles);
-        const x = center + length * Math.sin(base60ToRadians(base60Distance));
-        const y = center + length * -1 * Math.cos(base60ToRadians(base60Distance));
-        canvas.beginPath();
-        canvas.moveTo(center, center);
-        canvas.lineTo(x, y);
-        canvas.closePath();
-        canvas.stroke();
-      };
-
-      // draw second hand
       this.listenTo("time", (ev, time) => {
         canvas.clearRect(0, 0, diameter, diameter);
```

Comment:
This code assumes there’s a `canvas` variable, but there isn’t. I think we should probably use `this.canvas` everywhere.

Reply:
Done!